ktor-io/common/src/io/ktor/utils/io/Coroutines.kt

```diff
@@ -43,32 +43,6 @@ fun CoroutineScope.reader(
     block: suspend ReaderScope.() -> Unit
 ): ReaderJob = launchChannel(coroutineContext, ByteChannel(autoFlush), attachJob = true, block = block)
 
-@Deprecated("Use scope.reader instead")
-fun reader(
-    coroutineContext: CoroutineContext,
-    channel: ByteChannel,
-    parent: Job? = null,
-    block: suspend ReaderScope.() -> Unit
-): ReaderJob {
-    val newContext = if (parent != null) GlobalScope.newCoroutineContext(coroutineContext + parent)
-    else GlobalScope.newCoroutineContext(coroutineContext)
-
-    return CoroutineScope(newContext).reader(EmptyCoroutineContext, channel, block)
-}
-
-@Suppress("DEPRECATION")
-@Deprecated("Use scope.reader instead")
-fun reader(
-    coroutineContext: CoroutineContext,
-    autoFlush: Boolean = false, parent: Job? = null,
-    block: suspend ReaderScope.() -> Unit
-): ReaderJob {
-    val channel = ByteChannel(autoFlush)
-    return reader(coroutineContext, channel, parent, block).also {
-        channel.attachJob(it)
-    }
-}
-
```

Comment:
Why did we need this Reader/Writer jobs before and why they are not used anymore?

Reply:
That's deprecated API without `CoroutineScope` that we don't want to support